### PR TITLE
Yarn improvements

### PR DESCRIPTION
**Pull request by rmetzger**
Implementation for issue https://github.com/stratosphere/stratosphere/issues/473

The new tar.gz file has the following structure:

```
./examples
./examples/stratosphere-java-examples-0.5-SNAPSHOT-EnumTrianglesWithDegrees.jar
./examples/stratosphere-java-examples-0.5-SNAPSHOT-WebLogAnalysis.jar
./examples/stratosphere-java-examples-0.5-SNAPSHOT-EnumTrianglesOnEdgesWithDegrees.jar
./examples/stratosphere-scala-examples-0.5-SNAPSHOT-ConnectedComponents.jar
./examples/stratosphere-scala-examples-0.5-SNAPSHOT-TPCHQuery3.jar
./examples/stratosphere-scala-examples-0.5-SNAPSHOT-WordCount.jar
./examples/stratosphere-scala-examples-0.5-SNAPSHOT-KMeansForTest.jar
./examples/stratosphere-java-examples-0.5-SNAPSHOT-TPCHQuery3.jar
./examples/stratosphere-scala-examples-0.5-SNAPSHOT-KMeans.jar
./examples/stratosphere-java-examples-0.5-SNAPSHOT-DanglingPageRank.jar
./examples/stratosphere-java-examples-0.5-SNAPSHOT-ConnectedComponents.jar
./examples/stratosphere-java-examples-0.5-SNAPSHOT-KMeansIterative.jar
./examples/stratosphere-java-examples-0.5-SNAPSHOT-TeraSort.jar
./examples/stratosphere-scala-examples-0.5-SNAPSHOT-ComputeEdgeDegrees.jar
./examples/stratosphere-java-examples-0.5-SNAPSHOT-PairwiseSP.jar
./examples/stratosphere-java-examples-0.5-SNAPSHOT-EnumTrianglesRdfFoaf.jar
./examples/stratosphere-scala-examples-0.5-SNAPSHOT-EnumTrianglesOnEdgesWithDegrees.jar
./examples/stratosphere-java-examples-0.5-SNAPSHOT-WordCount.jar
./examples/stratosphere-java-examples-0.5-SNAPSHOT-KMeans.jar
./.version.properties
./tools
./tools/planVisualizer.html
./conf
./conf/log4j.properties
./conf/stratosphere-conf.yaml
./LICENSE.txt
./NOTICE.txt
./README.txt
./bin
./bin/config.sh
./bin/yarn-session.sh
./bin/stratosphere
./log
./lib
./lib/stratosphere-dist-0.5-SNAPSHOT-yarn-uberjar.jar
```

I will test this in the AWS Cloud and update the blog post.

### CODE REVIEW

**Comment by rmetzger** [comment]
Tested on amazon's servers.
I'll test the travis->s3 deployment. If it passes, I'll merge it to finally publish the blog post.

**Comment by fhueske** [comment]
Structure looks good to me. Nice!

File: stratosphere-addons/yarn/src/main/java/eu/stratosphere/yarn/Client.java [hunk 1]
```diff
@@ -337,9 +337,19 @@ public boolean accept(final File dir, final String name) {
...
+		
+		
 		 Runtime.getRuntime().addShutdownHook(new Thread() {
 		   @Override
 		   public void run() {
+		    try {
+		    	LOG.info("Killing the YARN instance.");
+				yarnClient.killApplication(appId);
+			} catch (YarnException e) {
+				e.printStackTrace();
```
Comment (StephanEwen) [inline stratosphere-addons/yarn/src/main/java/eu/stratosphere/yarn/Client.java]: Can you make this a log statement?

**Comment by StephanEwen** [comment]
Looks good. One minor comment, otherwise +1

**Comment by rmetzger** [comment]
Thanks.
I merged it.

**Pull request merged**